etc/init.d/zfs-import.in

```diff
@@ -0,0 +1,294 @@
+#!@SHELL@
+#
+# zfs-mount     This script will import/export zfs pools.
+#
+# chkconfig:    2345 01 99
+# description:  This script will import/export zfs pools during system
+#               boot/shutdown.
+#               It is also responsible for all userspace zfs services.
+# probe: true
+#
+### BEGIN INIT INFO
+# Provides:          zfs-import
+# Required-Start:    zfs-zed
+# Required-Stop:     zfs-zed
+# Default-Start:     S
+# Default-Stop:      0 1 6
+# X-Start-Before:    checkfs
+# X-Stop-After:      zfs-mount
+# Short-Description: Import ZFS pools
+# Description: Run the `zpool import` or `zpool export` commands.
+### END INIT INFO
+
+# Source the common init script
+. @sysconfdir@/zfs/common.init
+
+# ----------------------------------------------------
+
+depend()
+{
+	need zfs-zed
+}
+
+# Support function to get a list of all pools, separated with ';'
+find_pools()
+{
+	local CMD="$*"
+	local pools
+
+	pools=$($CMD 2> /dev/null | \
+		grep -E "pool:|^[a-zA-Z0-9]" | \
+		sed 's@.*: @@' | \
+		while read pool; do \
+		    echo -n "$pool;"
+		done)
+
+	echo "${pools%%;}" # Return without the last ';'.
+}
+
+# Import all pools
+do_import()
+{
+	local already_imported available_pools pool apools new_pools
+	local exception dir r RET=0
+
+	already_imported=$(find_pools "$ZPOOL" list -H -oname)
+	available_pools=$(find_pools "$ZPOOL" import)
+
+	# Just in case - seen it happen
+	if [ "$USE_DISK_BY_ID" = 'yes' ]; then
+		new_pools=$(find_pools "$ZPOOL" import -d /dev/disk/by-id)
```

Comment:
I would suggest adding a `-F` here to match the behavior when the module auto-imports the pool itself to fix #3434.